plugins/auth/contexts.js

```diff
@@ -17,14 +17,17 @@ module.exports = config => ({
         notes: 'Get all auth contexts',
         tags: ['api', 'auth', 'context'],
         handler: (request, reply) => {
-            const { scm } = request.server.app.userFactory;
-            const scmContexts = scm.getScmContexts();
+            const { pipelineFactory, userFactory } = request.server.app;
+            const scmContexts = userFactory.scm.getScmContexts();
             const contexts = [];
 
             scmContexts.forEach(scmContext => {
                 const context = {
                     context: scmContext,
-                    displayName: scm.getDisplayName({ scmContext })
+                    displayName: userFactory.scm.getDisplayName({ scmContext }),
+                    autoDeployKeyGeneration: pipelineFactory.scm.autoDeployKeyGenerationEnabled({
+                        scmContext
```

Comment:
is pipelineFactory.scm & userFactory.scm referring to same ?

Reply:
Yeah I guess I can use userFactory for both.